### PR TITLE
[Net] Add support for ipForward

### PR DESCRIPTION
**Pull request by way-zer**

### CODE REVIEW

File: core/src/mindustry/net/Administration.java [hunk 1]
```diff
@@ -479,6 +479,7 @@ private void load(){
         autosave("Whether the periodically save the map when playing.", false),
         autosaveAmount("The maximum amount of autosaves. Older ones get replaced.", 10),
         autosaveSpacing("Spacing between autosaves in seconds.", 60 * 5),
+        ipForward("Support ip forward behind proxy(don't enable if no proxy).", false),
```
Comment (joshuaptfan) [inline core/src/mindustry/net/Administration.java:482]:
```suggestion
        ipForward("Support IP forward behind proxy. Don't enable if no proxy.", false),
```

File: core/src/mindustry/net/ArcNetProvider.java [hunk 2]
```diff
@@ -81,6 +82,10 @@ public void received(Connection connection, Object object){
             @Override
             public void connected(Connection connection){
                 String ip = connection.getRemoteAddressTCP().getAddress().getHostAddress();
+                if(Config.ipForward.bool()){
+                    Log.debug("&bWait For IpForward: @", ip);
```
Comment (joshuaptfan) [inline core/src/mindustry/net/ArcNetProvider.java:86]:
```suggestion
                    Log.debug("&bWait For IP Forward: @", ip);
```

**Review by joshuaptfan — CHANGES_REQUESTED** [review]
Spelling/grammar.

**Review by Darkness6030 — APPROVED** [review]
Yes, it is a good feature

**Comment by bvcbvc** [comment]
this is going to make port fowarding easier i assume?

**Comment by way-zer** [comment]
> this is going to make port fowarding easier i assume?

not port forward, It is ip forward. 


> What does this actually do? Why are these changes necessary? 

There is a proxy like bungeecord but for mindustry. The proxy has the real ip address of client and send to tell the server behind proxy.
Reuse ConnectPacket is the simplest way to support ip forward in server.
To implement this in proxy, you need to send Connect Packet(with current address) after RegisterUDP.

> How would I test this?

May custom client or simulate receive Connect Packet.

**Comment by Anuken** [comment]
> There is a proxy like bungeecord but for mindustry. The proxy has the real ip address of client and send to tell the server behind proxy.
Reuse ConnectPacket is the simplest way to support ip forward in server.
To implement this in proxy, you need to send Connect Packet(with current address) after RegisterUDP.

I'm not familiar with bungeecord - what is the purpose of doing this?

> May custom client or simulate receive Connect Packet.

I don't understand what you mean by this.

**Comment by way-zer** [comment]
> I'm not familiar with bungeecord - what is the purpose of doing this?

like nginx?

> May custom client or simulate receive Connect Packet.
For ipForward disable, it has no change to cureent code.
For ipForward enable, you need to send Connect Packet (mock or from custom client)

**Pull request closed without merging**